akka-stream/src/main/scala/akka/stream/impl/UnfoldResourceSourceAsync.scala

```diff
@@ -102,7 +110,14 @@ import scala.util.control.NonFatal
     }
 
     private def createResource(): Unit = {
-      create().onComplete(createdCallback)
+      create().onComplete { resource ⇒
+        if (stoppedEarly) {
+          resource match {
+            case Success(r)  ⇒ close(r)
+            case Failure(ex) ⇒ throw ex
```

Comment:
Do we want to throw it or fail the stage with it? Is there a difference?

Reply:
The stage has already stopped, it cannot be failed.